src/services/user/hooks/handle-facebook-create.js

```diff
@@ -0,0 +1,52 @@
+
+const handleFacebookCreate = () => (hook) => {
+  // Pass through this hook if there's no facebook data
+  if (!hook.data.facebookId
+    || !hook.data.facebook
+    || !hook.data.facebook.email
+    || (!hook.data.facebook.name
+      && !hook.data.facebook.first_name
+      && !hook.data.facebook.last_name)
+    ) {
+    return hook;
+  }
+
+  const userService = hook.app.service('/users');
```

Comment:
Just see a user did like this in github issue. I think it makes sense. So, it decouples from database abstraction layer (mongoose).